Add Eclipse formatter and checks

SBRitter opened the pull request:
Summary:
- Add Eclipse formatter and compiler (same as lobid-organisations)
- Refactor code according to compiler warnings

Will resolve #31 

Note: for some reason, two commits by philboeselager from issue #25 that are already merged in the master (pull request: #33) are contained in this pull request, too. The commits corresponding to the two in the master are these: 3d858762046f7c33e7a48b12c7693d4ea8996a11 and 8107e506e411095acfa14c5d3ff6579b080c64a8

philboeselager committed:
Explicitly call "geodata" in path (#25)

philboeselager committed:
Fix route for Application.index() (#25)

SBRitter committed:
Merge branch 'master' of https://github.com/hbz/geodata

SBRitter committed:
Add settings with formatter and compiler checks (#31)

Same as in lobid-organisations

SBRitter committed:
Format classes according to new formatter (#31)

SBRitter committed:
Change code according to compiler checks (#31)

- Javadoc comments
- try-wth-resources statements

SBRitter committed:
Suppress javadoc warning on Application.java (#31)

fsteeg commented on .settings/.settings/org.eclipse.m2e.core.prefs:
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1
```
I think `org.eclipse.m2e.core.prefs` can be removed, it's no longer a Maven project.

fsteeg commented on .settings/.settings/org.scala-ide.sdt.core.prefs:
```diff
@@ -0,0 +1,6 @@
+scala.compiler.additionalParams=-encoding utf8
+deprecation=true
+unchecked=true
+scala.compiler.useProjectSettings=true
```
We should .gitignore `org.scala-ide.sdt.core.prefs`, which is generated by sbteclipse.

fsteeg commented on .settings/.settings/org.eclipse.core.resources.prefs:
```diff
@@ -0,0 +1,3 @@
+eclipse.preferences.version=1
+encoding//test/transformation/input/dbs.csv=ISO-8859-1
+encoding/<project>=UTF-8
```
This looks like a duplicate settings file (same for all in `.settings/.settings`).

fsteeg commented on .settings/org.eclipse.m2e.core.prefs:
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1
```
I think `org.eclipse.m2e.core.prefs` can be removed, it's no longer a Maven project.

fsteeg commented on .settings/org.scala-ide.play2.prefs:
```diff
@@ -0,0 +1,4 @@
+#Generated by sbt-plugin
+#Fri Jul 08 14:17:49 CEST 2016
+eclipse.preferences.version=1
+templateImports=import play.api.templates._\nimport play.api.templates.PlayMagic._\nimport models._\nimport controllers._\nimport java.lang._\nimport java.util._\nimport scala.collection.JavaConversions._\nimport scala.collection.JavaConverters._\nimport play.api.i18n._\nimport play.core.j.PlayMagicForJava._\nimport play.mvc._\nimport play.data._\nimport play.api.data.Field\nimport play.mvc.Http. ...
```
We should .gitignore `org.scala-ide.play2.prefs`, which is generated by sbteclipse.

fsteeg commented on .settings/org.scala-ide.sdt.core.prefs:
```diff
@@ -0,0 +1,6 @@
+scala.compiler.additionalParams=-encoding utf8
+deprecation=true
+unchecked=true
+scala.compiler.useProjectSettings=true
```
We should .gitignore `org.scala-ide.sdt.core.prefs`, which is generated by sbteclipse.

fsteeg commented on app/controllers/Application.java:
```diff
@@ -3,10 +3,12 @@
 import play.mvc.Controller;
 import play.mvc.Result;
 
+@SuppressWarnings("javadoc")
```
Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

fsteeg commented on app/controllers/Global.java:
```diff
@@ -7,6 +7,7 @@
 import play.GlobalSettings;
 import play.Logger;
 
+@SuppressWarnings("javadoc")
```
Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

fsteeg commented on app/controllers/geo/Constants.java:
```diff
@@ -1,5 +1,6 @@
 package controllers.geo;
 
+@SuppressWarnings("javadoc")
```
Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

SBRitter replied:
I agree! However, these constants are really self-explanatory, aren't they? That is, I can add the comments, but will it really add information?

fsteeg commented on app/controllers/geo/GeoElasticsearch.java:
```diff
@@ -12,67 +12,89 @@
+/**
+ * Configuration and creation of Elasticsearch index
+ */
 @SuppressWarnings("javadoc")
```
Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

fsteeg commented on app/controllers/geo/GeoInformator.java:
```diff
@@ -11,6 +11,11 @@
+ * Methods to trigger geo lookups on the index
+ *
+ */
+@SuppressWarnings("javadoc")
```
Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

fsteeg commented on app/controllers/geo/QueryHelpers.java:
```diff
@@ -13,54 +13,61 @@
+/**
+ * Helper methods for queries
+ */
+@SuppressWarnings("javadoc")
```
Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

SBRitter committed:
Delete duplicate settings files (#31)

SBRitter committed:
Delete .settings/org.eclipse.m2e.core.prefs (#31)

SBRitter committed:
Remove sbt generated settings files  and gitignore them (#31)

Remove these files and gitignore them:
.settings/org.scala-ide.play2.prefs
.settings/org.scala-ide.sdt.core.prefs

SBRitter committed:
Remove surpress javadoc warning annotations and add missing comments (#31)

fsteeg commented:
+1

SBRitter committed:
Change access modifiers of fields in class GeoElasticsearch (#31)
